### DISCUSSION TITLE
Hello, I haven't come to terms with this problem for days.

### DISCUSSION THREAD

**Discussion by babbioooooo** [Jyers/Marlin#527] [Q&A] [answered]
Hello, I haven't come to terms with this problem for days.  I installed jyers 1.3.0, I do the automatic bltouch leveling, I adjust the z-offset (-2.79) and everything is perfect, the problem arises when I start the print, I find myself with a much higher z-offset (I think  that goes to -1.5).  I tried to do a default reset and to recalibrate everything but the problem persists.  Unfortunately I am in the condition that to print something during printing I have to adjust the z-offset with manual microstep and above all by eye.  How can I fix it, believe me I have been trying for more than a week, I reset, I install again, but I can't.  thanks in advance for the reply.

**Comment by gothcha** [comment, marked as answer]
Did you store the settings after adjusting the Z-offset? When you do the z-offset should show -2.97 in the display on power-up.

**Reply by babbioooooo** [reply]
Of course, I have the habit of always pressing store setting for every change I make, however it shows 2.79 on the screen, even after switching off.

**Reply by gothcha** [reply]
Great, is it possible that the there is something being set in the slicer? Otherwise I can't think of why you should have this issue, it has never been an issue for me, but I am using 1.2.2 at present but I cannot believe it is an issue with the firmware.

**Reply by babbioooooo** [reply]
thanks for the kindness.  I am attaching the gcode of Cura 4.8, I am a beginner and this world fascinates me a lot.

**Reply by babbioooooo** [reply]
G-code
G92 E0 ; Reset Extruder
G28 ; Home all axes
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X0.1 Y20 Z0.3 F5000.0 ; Move to start position
G1 X0.1 Y200.0 Z0.3 F1500.0 E15 ; Draw the first line
G1 X0.4 Y200.0 Z0.3 F5000.0 ; Move to side a little
G1 X0.4 Y20 Z0.3 F1500.0 E30 ; Draw the second line
G92 E0 ; Reset Extruder
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X5 Y20 Z0.3 F5000.0 ; Move over to prevent blob squish

**Reply by gothcha** [reply]
It appears that you have not enabled the bed levelling. After the G28 you should add

M420 S1 Z2

This will enable the levelling and set the fade height to 2mm.

If you wish to do the levelling on every print you can replace the M420 command with a simple G29

Hope this helps

**Reply by gothcha** [reply]
I have assumed that you installed the BLTouch-3x3 or 5x5 firmware.

Note that if you have used the UBL-BLTouch then the G codes are different!

**Reply by babbioooooo** [reply]
Sorry, I forgot to write it down, I installed 10x10 ubl

**Reply by gothcha** [reply]
OK for the UBL then you need to insert after G28

G29 A ; Activate UBL
G29 L0 ; Load the stored mesh slot 0 not essential if it has been loaded from menu
G29 J ; Probe 3 points to tilt mesh

The G29 J is not essential unless the bed has tilted since the mesh was generated.

Also I presume that you stored the mesh calibration after you ran it from the levelling menu? And that you set the BLTouch offsets in the advanced menu (typically -42,-10)

**Reply by babbioooooo** [reply]
Perfect, I immediately try a mini print and let you know, yes after the creation of the mesh I always do "save mesh"

**Reply by babbioooooo** [reply]
> OK per l'UBL, quindi è necessario inserire dopo G28
> 
> G29 A; Attiva UBL 
> G29 L0; Carica la mesh slot 0 memorizzata non indispensabile se è stata caricata dal menù 
> G29 J; Sonda 3 punti per inclinare la rete
> 
> Il G29 J non è essenziale a meno che il letto non si sia inclinato da quando è stata generata la rete.
> 
> Inoltre presumo che tu abbia memorizzato la calibrazione della mesh dopo averla eseguita dal menu di livellamento? E che imposti gli offset BLTouch nel menu avanzato (in genere -42, -10)

First of all, thank you very much for your availability, now the z has lowered with the new gcode and I am very happy because I believe that today I solve the problem, but now the only problem that remains is: the z-offset is lower, in fact it does not print  because it is too close to the plate.

**Reply by babbioooooo** [reply]
> OK per l'UBL, quindi è necessario inserire dopo G28
> 
> G29 A; Attiva UBL 
> G29 L0; Carica la mesh slot 0 memorizzata non indispensabile se è stata caricata dal menù 
> G29 J; Sonda 3 punti per inclinare la rete
> 
> Il G29 J non è essenziale a meno che il letto non si sia inclinato da quando è stata generata la rete.
> 
> Inoltre presumo che tu abbia memorizzato la calibrazione della mesh dopo averla eseguita dal menu di livellamento? E che imposti gli offset BLTouch nel menu avanzato (in genere -42, -10)

I did another test, by removing the G29 J the nozzle remains high for me, instead by inserting the G29 J the nozzle sticks to the plate at 0mm.  what can it be?

**Reply by gothcha** [reply]
The G29 J should just run the 3 point levelling, does it do this correctly? It will say 'Tilting point 1/3' ....

Sorry I am not sure what else to suggest, maybe someone else has some solution??

If you are using the Ender3V2 with the glass plate then the bed is already very flat. I found that manually levelling the bed and using the standard 3x3 firmware is generally more than good enough and very simple to use. If this is not working correctly then the UBL firmware will not either.

**Reply by babbioooooo** [reply]
> Il G29 J dovrebbe eseguire solo il livellamento a 3 punti, lo fa correttamente? Dirà "Punto di inclinazione 1/3" ....
> 
> Scusa non sono sicuro di cos'altro suggerire, forse qualcun altro ha qualche soluzione ??
> 
> Se stai usando Ender3V2 con la lastra di vetro, il letto è già molto piatto. Ho scoperto che livellare manualmente il letto e utilizzare il firmware 3x3 standard è generalmente più che sufficiente e molto semplice da usare. Se questo non funziona correttamente, anche il firmware UBL non funzionerà.

I installed the 3x3 bltouch and now it seems to be fine, but with the z-offset I have to keep lower and lower, I find it very strange that with ubl it gives this problem.  let's see if jyers can explain why.  in the meantime I really wanted to thank you from the heart for the advice you gave me.
